Tests/SonarScanner.MSBuild.PreProcessor.Test/WebClientDownloaderTest.cs

```diff
@@ -144,6 +150,112 @@ public void MultipleDisposeCallsNotFailing()
             testDownloader.IsDisposedCalled.Should().BeTrue();
         }
 
+        [TestMethod]
+        public async Task DownloadStream_Success()
+        {
+            const string testContent = "test content";
+            var logger = new TestLogger();
+            var httpClient = MockHttpClient(new HttpResponseMessage { StatusCode = HttpStatusCode.OK, Content = new StringContent(testContent) });
+
+            var sut = new WebClientDownloader("username", "password", logger, httpClient: httpClient);
+            var stream = await sut.DownloadStream(new Uri("https://test.com"));
+
+            var text = await new StreamReader(stream).ReadToEndAsync();
+            text.Should().Be(testContent);
+
+            logger.AssertDebugLogged("Downloading from https://test.com/...");
+        }
+
+        [TestMethod]
+        public async Task DownloadStream_Fail_ForbiddenResponse_WithLog()
+        {
+            const string testContent = "test content";
+            var logger = new TestLogger();
+            var httpClient = MockHttpClient(new HttpResponseMessage { StatusCode = HttpStatusCode.Forbidden, Content = new StringContent(testContent) });
+
+            var sut = new WebClientDownloader("username", "password", logger, httpClient: httpClient);
```

Comment:
What about something like this, to remove duplicated scaffolding?
```suggestion
            var sut = CreateWebClientDownloader(logger, new HttpResponseMessage { StatusCode = HttpStatusCode.Forbidden, Content = new StringContent(testContent));
```

Reply:
I've extracted also the `HttpResponseMessage`. The tests got really simple now :)

Reply:
🎉